DataSynapseApi/ArmoniK.DevelopmentKit.GridServer.Client/Service.cs

```diff
@@ -249,6 +267,67 @@ public string Submit(string methodName, byte[] argument, IServiceInvocationHandl
                     handler);
     }
 
+    private byte[] ActiveGetResult(params string[] taskId)
+    {
+      return ActiveGetResults(taskId).Single().Item2;
+    }
+
+    private IEnumerable<Tuple<string, byte[]>> ActiveGetResults(IEnumerable<string> taskIds)
+    {
+      var ids      = taskIds.ToList();
+      var missing  = ids;
+      var results  = new List<Tuple<string, byte[]>>();
+      var holdPrev = 0;
+      var waitInSeconds = new List<int>
+      {
+        10,
+        1000,
+        5000,
+        10000,
+        20000,
+      };
+      var idx = 0;
+
+      Logger.BeginPropertyScope(("SessionId", SessionId), ("Function", "ActiveGetResults"));
+
+      while (missing.Count != 0)
+      {
+        var buckets = missing.Batch(10000).ToList();
+
+        buckets.ForEach(bucket =>
+        {
+          var partialResults = SessionService.TryGetResults(bucket);
+
+          var listPartialResults = partialResults.ToList();
+
+          if (listPartialResults.Count() != 0)
+          {
+            results.AddRange(listPartialResults);
+          }
+
+          missing = missing.Where(x => listPartialResults.ToList().All(rId => rId.Item1 != x)).ToList();
```

Comment:
This is accidentally quadratic! Please use an hash set/map to know what are the missing ids, or use an intersection of sorted lists.

This is also a problem in `UnifiedApi/Client/Services/Submitter/Service.cs` line 295

Reply:
fixed